Documentation/DesignGuidelines.md

```diff
@@ -8,26 +8,72 @@ This document assumes basic familiarity
 with the features of ReactiveCocoa. The [Framework Overview][] is a better
 resource for getting up to speed on the functionality provided by RAC.
 
-**[The Signal contract](#the-signal-contract)**
+1. [The `Event` contract](#the-event-contract)
+1. [The `Signal` contract](#the-signal-contract)
+1. [The `SignalProducer` contract](#the-signalproducer-contract)
+1. [Best practices](#best-practices)
+1. [Implementing new operators](#implementing-new-operators)
 
-**[The SignalProducer contract](#the-signalproducer-contract)**
+## The `Event` contract
 
-**[Best practices](#best-practices)**
+Events are fundamental to ReactiveCocoa. Signals and signal producers both send
+events, and may be collectively called “event streams.”
 
-**[Implementing new operators](#implementing-new-operators)**
+Event streams must conform to the following grammar:
 
+```
+Next* (Interrupted | Error | Completed)?
+```
 
-## The Signal contract
+This states that an event stream consists of:
 
+ 1. Any number of `Next` events
+ 1. Optionally followed by one terminating event, which is any of `Interrupted`, `Error`, or `Completed`
 
-## The SignalProducer contract
+After a terminating event, no other events will be received.
 
+#### `Next`s provide values or indicate the occurrence of events
+#### Errors behave like exceptions and propagate immediately
+#### Interruption cancels outstanding work and usually propagates immediately
+#### Completion indicates success
+#### Events are serial
+#### Events cannot be sent recursively
+#### Events are sent synchronously by default
+
+## The `Signal` contract
+
+#### Signals start work when instantiated
+#### Observing a signal does not have side effects
+#### All observers of a signal see the same events at the same time
+#### Signals are retained until a terminating event occurs
+#### Terminating events dispose of signal resources
+
+## The `SignalProducer` contract
+
+#### Signal producers start work on demand by creating signals
+#### Each produced signal may send different events at different times
+#### Disposing of a produced signal will interrupt it
+#### Signal operators can be lifted to apply to signal producers
 
 ## Best practices
 
+#### Indent signal and producer chains consistently
+#### Process only as many values as needed
+#### Deliver events onto a known scheduler
+#### Switch schedulers in as few places as possible
+#### Capture side effects within signal producers
```

Comment:
e.g. if you have a method returning a `SignalProducer`, do the work in the closure passed to the constructor, not just inside the method itself.